### PR TITLE
Added flag to control the network policy creation

### PR DESCRIPTION
**Pull request by Rubemlrm**
### Description
This change was relative to a new flag that will validate the need to create a network policy. I had an use case where the user accounts can't create network policies.
  
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
Hi @Rubemlrm could you create an issues related to this?

Also, due to 1.2.1 PR #164 is merged, you need to pull the latest changes, use 1.5.5 version number, and resolve conflicts.

If you can also remove the extra spaces you introduced in changelog, that will be great.

Thanks.

**Comment by Rubemlrm** [comment]
> Hi @Rubemlrm could you create an issues related to this?
> 
> Also, due to 1.2.1 PR #164 is merged, you need to pull the latest changes, use 1.5.5 version number, and resolve conflicts.
> 
> If you can also remove the extra spaces you introduced in changelog, that will be great.
> 
> Thanks.

Hi @peterzhuamazon, created the issue #176 for this PR. I will do the changes that you recommended.

**Comment by Rubemlrm** [comment]
@peterzhuamazon i think that now it's all fine. I've merged the main branch and bumped versions.

File: charts/opensearch/CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,22 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+### Added
+
+- Added `create` flag into the `networkPolicy` resource to enable the creation of a network policy as request, for the cases where an user can't create network policies but want to use this chart.
+
+### Changed
+
+### Deprecated
+
+### Removed
+
```
Comment (peterzhuamazon) [inline charts/opensearch/CHANGELOG.md:27]: Nitpik: These extra spaces can be removed.

File: charts/opensearch/README.md [hunk 2]
```diff
@@ -2,10 +2,10 @@
...
 This Helm chart installs [OpenSearch](https://github.com/opensearch-project/OpenSearch) with configurable TLS, RBAC and much more configurations. This chart caters a number of different use cases and setups.
 
-- [Requirements](#requirements)
-- [Installing](#installing)
-- [Uninstalling](#uninstalling)
-- [Configuration](#configuration)
+  - [Requirements](#requirements)
+  - [Installing](#installing)
+  - [Uninstalling](#uninstalling)
+  - [Configuration](#configuration)
```
Comment (peterzhuamazon) [inline charts/opensearch/README.md:8]: Additional spaces introduced.

File: charts/opensearch/README.md [hunk 3]
```diff
@@ -66,6 +66,7 @@ helm uninstall my-release
 | `minimumMasterNodes`               | The value for `discovery.zen.minimum_master_nodes`                                                                                 | `2`                                             |
 | `nameOverride`                     | Overrides the `clusterName` when used in the naming of resources                                                                                                                                                                                          | `""`                                            |
 | `networkHost`                      | Value for the `network.host OpenSearch setting`                                                                                                                                                                                                      | `0.0.0.0`                                       |
+| `networkPolicy.create`                      | Enable network policy creation for OpenSearch | `false`
```
Comment (peterzhuamazon) [inline charts/opensearch/README.md:69]: You have additional spaces leading to `| Enable......` as well.

**Comment by peterzhuamazon** [comment]
I think your IDE is adding extra spaces everywhere, can you take a look @Rubemlrm and address my comments?
Thanks.

File: charts/opensearch/README.md (hunk 3 above)
Comment (Rubemlrm) [reply charts/opensearch/README.md]: on this one, i have a question, the ideia is  to ident the '|' after the `enable` like is on the line above?

**Comment by Rubemlrm** [comment]
> I think your IDE is adding extra spaces everywhere, can you take a look @Rubemlrm and address my comments? Thanks.

Just updated the code, i had a question relative to the comment on the parameters list, it's to align the '|' before Enable with the upper line ? Or was something else?

thanks a lot for the help and feedback

**Comment by peterzhuamazon** [comment]
Stays on 1.5.5.

File: charts/opensearch/README.md (hunk 3 above)
Comment (peterzhuamazon) [reply charts/opensearch/README.md]: I think your new application is good. Thanks.

**Review by TheAlgo — APPROVED** [review]

**Comment by TheAlgo** [comment]
Skipping @DandyDeveloper 👍 since he already approved #159

**Comment by TheAlgo** [comment]
@DandyDeveloper Please have a look when free. Thanks

**Review by DandyDeveloper — APPROVED** [review]
Good change, should have been conditional from the start. Thank you @Rubemlrm

**Comment by TheAlgo** [comment]
Thanks @Rubemlrm for the PR. Merging it.

**Pull request merged**